src/pages/blog/2024-08-14-semantic-nullability-for-application-developers.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: "Semantic Nullability For Application Developers"
+tags: ["spec"]
+date: 2024-09-12
+byline: Alex Reilly
+---
+
+# Semantic Nullability
+
+> This blog post is directed at application developers using GraphQL. If you are a library author, you should read the more detailed feature spec instead.
+
+Today we're providing a progress update from the Nullablility Working Group on Semantic Nullability which is our new approach to fixing GraphQL's nullability issues.
+
+GraphQL has some fundamental problems, and to talk about them, we first have to talk about GraphQL's type system. GraphQL allows you to define a schema, and to do that many developers write a document in Schema Definition Language, or SDL. A SDL document may look like this
+
+```graphql
+type User { 
+  id: ID! 
+  name: String!
+  age: Int
+  posts: [Post] 
+}
+```
+
+One thing to make clear: this is not like using a type system in a compiled language (TS, Swift, Kotlin) where the type system makes compile-time guarantees about behavior at runtime. Rather you can think of GraphQL's type system as a "runtime" type system. You can trust that `age` will be an `Int`, because at runtime, GraphQL will assert that it is a `Int`, and throw an error if it is not. All types effectively represent a typecast, or a type assertion.
```

Comment:
> You can trust that `age` will be an `Int`, because at runtime, GraphQL will assert that it is a `Int`, and throw an error if it is not.

`age` is nullable here, so it could be `null` as well, and won't throw for a `null` value. Maybe `name` would be a better example?